oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/record-requests/components/RequestModal.jsx

```diff
@@ -105,32 +84,33 @@ export const RequestModal = ({ request, requestTypes, requestModalType, isEventM
       headers: { 'Content-Type': 'application/json' }
     })
       .then(response => {
-        console.log(response);
-        fetchUpdated(record.links?.requests, (requests) => { setRequests(sortByStatusCode(requests)); });
-        setModalOpen(false);
-        formik.resetForm();
+        if (isMounted.current) {
+          setModalOpen(false);
+          formik.resetForm();
+        }
+        fetchNewRequests();
       })
       .catch(error => {
-        console.log(error);
         setError(error);
       })
       .finally(() => {
-        formik.setSubmitting(false);
+        isMounted.current && formik.setSubmitting(false);
       });
   }
 
   const createAndSubmitRequest = async () => {
     try {
       const createdRequest = await callApi(request.links.actions?.create, 'post', formik.values, true);
       await callApi(createdRequest.data?.links?.actions?.submit, 'post', {}, true);
-      fetchUpdated(record.links?.requests, (requests) => { setRequests(sortByStatusCode(requests)); });
-      setModalOpen(false);
-      formik.resetForm();
+      if (isMounted.current) {
```

Comment:
This shouldn't be done via ref, useEffect should be used for running effects on mount

Reply:
Removed unnecessary useIsMounted